python/ray/tune/schedulers/pbt.py

```diff
@@ -433,6 +429,25 @@ def on_trial_result(self, trial_runner: "trial_runner.TrialRunner",
             # the paused trials.
             return TrialScheduler.PAUSE
 
+    def _save_trial_state(self, state: PBTTrialState, time: int, result: Dict,
```

Comment:
We can merge first, given how much work has gone into this PR. However, we will later want to make sure this (and _get_new_config) are public interfaces, so to prevent accidental breakage.